spec.md

```diff
@@ -3713,3 +3713,163 @@ that was stored and it is removed from the array.
 
 Ping array is used in many places in toxcore to efficiently keep track of sent
 packets.
+
+# State Format
+
+The reference Tox implementation uses a custom binary format to save the state
+of a Tox client between restarts. This format is far from perfect and will be
+replaced eventually. For the sake of maintaining compatibility down the road,
+it is documented here.
+
+The binary encoding of all integer types in the state format is a fixed-width
+byte sequence with the integer encoded in Little Endian.
+
+Length | Contents
+------ | --------
+`4`    | Zeroes
+`4`    | `uint32_t` (0x15ED1B1F)
+`?`    | List of sections
+
+## Sections
+
+The core of the state format consists of a list of sections. Every section has
+its type and length specified at the beginning. In some cases, a section only
+contains one item and thus takes up the entire length of the section. This is
+denoted with '?'.
+
+Length   | Contents
+-------- | --------
+`4`      | `uint32_t` Length of this section
+`2`      | `uint16_t` Section type
+`2`      | `uint16_t` (0x01CE)
+`?`      | Section
+
+Section types:
+
+Name          | Value
+------------- | -----
+NospamKeys    | 1
+DHT           | 2
+Friends       | 3
+Name          | 4
+StatusMessage | 5
+Status        | 6
+TcpRelays     | 10
+PathNodes     | 11
+EOF           | 255
+
+Not every section listed above is required to be present in order to restore
+from a state file. Only NospamKeys is required.
+
+### Nospam and Keys
+
+Length | Contents
+------ | --------
+`4`    | `uint32_t` Nospam
+`32`   | Long term public key
+`32`   | Long term secret key
+
+### DHT
```

Comment:
MD024 Multiple headers with the same content

Please choose a different name or qualify it (or the existing one).